net-vpn/i2p/i2p-2.3.0-r1.ebuild

```diff
@@ -0,0 +1,291 @@
+	find -name '*_iw.*' -exec rename --no-overwrite _iw. _he. {} \; ||
+		die 'fix some locale names'
+}
+
+src_configure() {
+	java-ant-2_src_configure
+
+	# deamon shouldn't start GUI
+	sed -i 's|\(clientApp.4.startOnLoad\)=true|\1=false|' \
+		installer/resources/clients.config ||
+		die 'avoid auto starting browser'
+
+	# yep, that's us
+	echo "build.built-by=Gentoo" >> override.properties ||
+		die 'bragging failed'
+}
+
+src_compile() {
+	java-pkg-2_src_compile
+
+	local compile_lib
+	compile_lib() {
+		local name="${1}"
+		local file="${2}"
+		shift 2
+
+		"$(tc-getCC)" "${@}" ${CFLAGS} $(java-pkg_get-jni-cflags) \
+			${LDFLAGS} -shared -fPIC "-Wl,-soname,lib${name}.so" \
+			"${file}" -o "lib${name}.so"
+	}
+
+	cd "${S}/core/c/jbigi/jbigi" || die "unable to cd to jbigi"
+	compile_lib jbigi src/jbigi.c -Iinclude -lgmp ||
+		die "unable to build jbigi"
+
+	if use amd64 || use x86; then
+		cd "${S}/core/c/jcpuid" || die "unable to cd to jcpuid"
+		compile_lib jcpuid src/jcpuid.c -Iinclude ||
+			die "unable to build jcpuid"
+	fi
+}
+
+src_test() {
+	# avoid rebuilding
+	sed -e '/<delete dir=".\/build" \/>/d' -i core/java/build.xml ||
+		die 'avoid building twice'
+
+	# halt on error
+	find -name build.xml \
+		-execdir sed -e 's/<junit /\0haltonerror="yes" /' -i {} + ||
+		die 'ensure test failures propagate'
+
+	EANT_GENTOO_CLASSPATH+=",hamcrest,junit-4,mockito-4"
+	java-pkg-2_src_test
+}
+
+src_install() {
+	# install basic documentation
+	einstalldocs
+	doman installer/resources/man/eepget.*
+
+	# install main files
+	java-pkg_doso core/c/jbigi/jbigi/libjbigi.so
+	if use amd64 || use x86; then
+		java-pkg_doso core/c/jcpuid/libjcpuid.so
+	fi
+	cd "${S}/pkg-temp" || die 'unable to change dir to built artifacts'
+	java-pkg_dojar lib/*.jar
+	java-pkg_dowar webapps/*.war
+
+	# install shared
+	insinto /usr/share/i2p
+	doins blocklist.txt hosts.txt {clients,i2p*}.config
+	doins -r certificates docs eepsite geoip scripts
+
+	# install daemons
+	newinitd "${FILESDIR}/i2p.init" i2p
+	systemd_dounit "${FILESDIR}/i2p.service"
+
+	# setup dirs
+	keepdir /var/log/i2p /var/lib/i2p
+	fowners i2p:i2p /var/lib/i2p /var/log/i2p
+
+	# create own launchers
+	java-pkg_dolauncher i2prouter --main net.i2p.router.Router --jar i2p.jar \
+		--pwd "${EPREFIX}/usr/share/i2p" \
+		--java_args "\
+			-Di2p.dir.config=${EPREFIX}/var/lib/i2p \
+			-Di2p.dir.log=${EPREFIX}/var/log/i2p \
+			-DloggerFilenameOverride=${EPREFIX}/var/log/i2p/router-@"
+	java-pkg_dolauncher eepget --main net.i2p.util.EepGet --jar i2p.jar
+}
+
+pkg_postinst() {
+	local i2pdir="${EPREFIX}/var/lib/i2p"
+
+	[ -d "${i2pdir}/app" -a -d "${i2pdir}/config" -a -d "${i2pdir}/router" ] || return
+
+	elog "Separated user directories is not fully supported by upstream."
+	elog "${i2pdir}/{app,config,router} will be merged"
+	elog "in ${i2pdir} accordingly."
+
+	ebegin "Migrating"
+	rm -fr "${i2pdir}/config/addressbook" # prefer router's addressbook
```

Comment:
Let's perhaps mv it to .bak instead, just in case?

Reply:
hum, only `config/addressbook` you mean or the whole i2pdir? the latter might be quite big to keep as apps also store their data in $i2pdir/apps

Reply:
I was thinking the whole thing but I suppose it's OK.